canvases/dark-tree-sitter-demo/main.dark

```diff
@@ -0,0 +1,375 @@
+      </div>
+    </div>
+    <script src="/static/tree-sitter/tree-sitter.js" />
+    <script>console.log('NOT SURE WHY I NEED THIS')</script>
+    <script>
+      function simplifyTree(cursor) {
+        let children = [];
+
+        if (cursor.gotoFirstChild()) {
+          do {
+            children.push(simplifyTree(cursor));
+          } while (cursor.gotoNextSibling());
+
+          cursor.gotoParent();
+        }
+
+        return {
+          typ: cursor.nodeType,
+          text: cursor.nodeText,
+          fieldName: cursor.currentFieldName(),
+          children: children,
+        };
+      }
+
+      const Parser = window.TreeSitter;
+
+      Parser.init().then(async () => {
+        const parser = new Parser;
+
+        const inputTextarea = document.getElementById("input-textarea");
+
+        const Darklang = await Parser.Language.load("/static/tree-sitter/tree-sitter-darklang.wasm");
+        parser.setLanguage(Darklang);
+
+        async function syntaxTreeToUserFunction(tree) {
+          try {
+            const response = await fetch("/parse-user-function", {
+              method: "POST",
+              headers: { "Content-Type": "application/json" },
+              body: JSON.stringify(tree)
+            });
+
+            if (!response.ok) {
+              const err = await response.text();
+              throw new Error("TODO", err);
+            }
+
+            const exprJson = await response.text(); // or maybe .json()?
+
+            document.getElementById("right-textarea").value = exprJson;
+
+            // TODO: now what? maybe eval via WASM?
+            // TODO: the WASM part; now that we have a user function,
+            //
+            // - load it into wasm
+            //   DotNet.invokeMethod("Wasm", "LoadExpr", exprJson);
+            //
+            // - run the fn (with some inputs)
+            //   DotNet.invokeMethod("Wasm", "EvalExpr");
+          } catch (error) {
+            console.error("Error:", error);
+          }
+        }
+
+        function sourceCodeToSyntaxTree () {
+          const darkSourceCode = document.getElementById("input-textarea").value;
+          const darkTree = parser.parse(darkSourceCode);
+          window.tree = darkTree;
+
+          let simpleTree = simplifyTree(darkTree.rootNode.walk());
+
+          document.getElementById("syntax-textarea").value = JSON.stringify(simpleTree, null, 2);
+
+          syntaxTreeToUserFunction(simpleTree);
+        }
+
+        inputTextarea.addEventListener("input", () => {
+          sourceCodeToSyntaxTree();
+        });
+        sourceCodeToSyntaxTree();
+      });
+    </script>
+  </body>
+</html>
+  """
+
+  Http.responseWithHeaders (String.toBytes body) (Dict.fromListOverwritingDuplicates [("Content-Type", "text/html")]) 200
+
+
+let gid (): Int =
+  Int.random_v1 0 9223372036854775807
+
+
+let err (msg: String): Unit =
+  // TODO: maybe something with the msg?
+  let _ = 1.0/0.0
+  ()
```

Comment:
It turns out that this ends up just returning `()`, without causing a failure like I expected.